app/main.py

```diff
@@ -1,8 +1,35 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def set_spouse(self, spouse_name):
+        if spouse_name:
+            spouse = Person.people.get(spouse_name)
+            if spouse:
+                if hasattr(self, "wife") or hasattr(self, "husband"):
+                    return  # Уже установлен супруг
+                if self.name not in ["Rachel", "Monica"]:
+                    self.wife = spouse
+                    spouse.husband = self
+                else:
+                    self.husband = spouse
+                    spouse.wife = self
+
+
+def create_person_list(data):
+    # Создание экземпляров Person
+    persons = [Person(item["name"], item["age"]) for item in data]
+
+    # Установка супругов
+    for item in data:
+        person = Person.people[item["name"]]
```

Comment:
The line `person = Person.people[item["name"]]` assumes that the person will always be found in the `people` dictionary. If the person is not found, this will raise a `KeyError`. Consider using `get` with a default value or adding error handling to manage this case.